### ISSUE TITLE
System.MissingMethodException - Initial Page Load

### ISSUE THREAD

**Issue by glorioustephan** [freshlogic/MongoDB.Web#16]
<strong>After placing the changes in the web.config, I am receiving this error when running the site.  I am running MVC3 with .NET 4.5.  I pulled this project and the official drivers from NuGet.  Any tips would be helpful. </strong>

<hr />

<h3>Method not found: 'MongoDB.Driver.Builders.QueryComplete MongoDB.Driver.Builders.Query.EQ(System.String, MongoDB.Bson.BsonValue)'.</h3>

<strong>Description: </strong>An unhandled exception occurred during the execution of the current web request. Please review the stack trace for more information about the error and where it originated in the code. 

<strong>Exception Details: </strong> System.MissingMethodException: Method not found: 'MongoDB.Driver.Builders.QueryComplete MongoDB.Driver.Builders.Query.EQ(System.String, MongoDB.Bson.BsonValue)'.

<strong>Source Error: </strong> 

An unhandled exception was generated during the execution of the current web request. Information regarding the origin and location of the exception can be identified using the exception stack trace below.

<strong>Stack Trace: </strong> 

<code>
[MissingMethodException: Method not found: 'MongoDB.Driver.Builders.QueryComplete MongoDB.Driver.Builders.Query.EQ(System.String, MongoDB.Bson.BsonValue)'.]
   MongoDB.Web.Providers.MongoDBOutputCacheProvider.Get(String key) in C:\MongoDB.Web\MongoDB.Web\Providers\MongoDBOutputCacheProvider.cs:40
   System.Web.Caching.OutputCache.Get(String key) +56
   System.Web.Caching.OutputCacheModule.OnEnter(Object source, EventArgs eventArgs) +9701754
   System.Web.SyncEventExecutionStep.System.Web.HttpApplication.IExecutionStep.Execute() +136
   System.Web.HttpApplication.ExecuteStep(IExecutionStep step, Boolean& completedSynchronously) +69
</code>

**Comment by MadsFSorensen** [comment]
Getting det same error 'MongoDB.Driver.Builders.QueryComplete MongoDB.Driver.Builders.Query.EQ(System.String, MongoDB.Bson.BsonValue)'

Server Error in '/' Application.

Method not found: 'MongoDB.Driver.Builders.QueryComplete MongoDB.Driver.Builders.Query.EQ(System.String, MongoDB.Bson.BsonValue)'.

Description: An unhandled exception occurred during the execution of the current web request. Please review the stack trace for more information about the error and where it originated in the code. 

Exception Details: System.MissingMethodException: Method not found: 'MongoDB.Driver.Builders.QueryComplete MongoDB.Driver.Builders.Query.EQ(System.String, MongoDB.Bson.BsonValue)'.

Source Error: 

Line 86:                // Attempt to register the user
Line 87:                MembershipCreateStatus createStatus;
Line 88:                Membership.CreateUser(model.UserName, model.Password, model.Email, passwordQuestion: null, passwordAnswer: null, isApproved: true, providerUserKey: null, status: out createStatus);
Line 89: 
Line 90:                if (createStatus == MembershipCreateStatus.Success)

Source File: D:\MyGit\MzM.SearchWordArticles\MzM.SearchWordArticles.Web\Controllers\AccountController.cs    Line: 88 

Stack Trace: 

[MissingMethodException: Method not found: 'MongoDB.Driver.Builders.QueryComplete MongoDB.Driver.Builders.Query.EQ(System.String, MongoDB.Bson.BsonValue)'.]
   MongoDB.Web.Providers.MongoDBMembershipProvider.GetUser(String username, Boolean userIsOnline) in C:\MongoDB.Web\MongoDB.Web\Providers\MongoDBMembershipProvider.cs:298
   MongoDB.Web.Providers.MongoDBMembershipProvider.CreateUser(String username, String password, String email, String passwordQuestion, String passwordAnswer, Boolean isApproved, Object providerUserKey, MembershipCreateStatus& status) in C:\MongoDB.Web\MongoDB.Web\Providers\MongoDBMembershipProvider.cs:154
   System.Web.Security.Membership.CreateUser(String username, String password, String email, String passwordQuestion, String passwordAnswer, Boolean isApproved, Object providerUserKey, MembershipCreateStatus& status) +207
   MzM.SearchWordArticles.Web.Controllers.AccountController.Register(RegisterModel model) in D:\MyGit\MzM.SearchWordArticles\MzM.SearchWordArticles.Web\Controllers\AccountController.cs:88
   lambda_method(Closure , ControllerBase , Object[] ) +162
   System.Web.Mvc.ActionMethodDispatcher.Execute(ControllerBase controller, Object[] parameters) +17
   System.Web.Mvc.ReflectedActionDescriptor.Execute(ControllerContext controllerContext, IDictionary`2 parameters) +205
   System.Web.Mvc.ControllerActionInvoker.InvokeActionMethod(ControllerContext controllerContext, ActionDescriptor actionDescriptor, IDictionary`2 parameters) +27
   System.Web.Mvc.Async.<>c__DisplayClass42.<BeginInvokeSynchronousActionMethod>b__41() +28
   System.Web.Mvc.Async.<>c__DisplayClass8`1.<BeginSynchronous>b__7(IAsyncResult _) +12
   System.Web.Mvc.Async.WrappedAsyncResult`1.End() +57
   System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethod(IAsyncResult asyncResult) +50
   System.Web.Mvc.Async.<>c__DisplayClass39.<BeginInvokeActionMethodWithFilters>b__33() +58
   System.Web.Mvc.Async.<>c__DisplayClass4f.<InvokeActionMethodFilterAsynchronously>b__49() +237
   System.Web.Mvc.Async.<>c__DisplayClass37.<BeginInvokeActionMethodWithFilters>b__36(IAsyncResult asyncResult) +12
   System.Web.Mvc.Async.WrappedAsyncResult`1.End() +57
   System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethodWithFilters(IAsyncResult asyncResult) +50
   System.Web.Mvc.Async.<>c__DisplayClass2a.<BeginInvokeAction>b__20() +24
   System.Web.Mvc.Async.<>c__DisplayClass25.<BeginInvokeAction>b__22(IAsyncResult asyncResult) +126
   System.Web.Mvc.Async.WrappedAsyncResult`1.End() +57
   System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeAction(IAsyncResult asyncResult) +45
   System.Web.Mvc.<>c__DisplayClass1d.<BeginExecuteCore>b__18(IAsyncResult asyncResult) +14
   System.Web.Mvc.Async.<>c__DisplayClass4.<MakeVoidDelegate>b__3(IAsyncResult ar) +25
   System.Web.Mvc.Async.WrappedAsyncResult`1.End() +62
   System.Web.Mvc.Controller.EndExecuteCore(IAsyncResult asyncResult) +61
   System.Web.Mvc.Async.<>c__DisplayClass4.<MakeVoidDelegate>b__3(IAsyncResult ar) +25
   System.Web.Mvc.Async.WrappedAsyncResult`1.End() +62
   System.Web.Mvc.Controller.EndExecute(IAsyncResult asyncResult) +49
   System.Web.Mvc.Controller.System.Web.Mvc.Async.IAsyncController.EndExecute(IAsyncResult asyncResult) +10
   System.Web.Mvc.<>c__DisplayClassb.<BeginProcessRequest>b__4(IAsyncResult asyncResult) +28
   System.Web.Mvc.Async.<>c__DisplayClass4.<MakeVoidDelegate>b__3(IAsyncResult ar) +25
   System.Web.Mvc.Async.WrappedAsyncResult`1.End() +62
   System.Web.Mvc.<>c__DisplayClasse.<EndProcessRequest>b__d() +50
   System.Web.Mvc.SecurityUtil.<GetCallInAppTrustThunk>b__0(Action f) +7
   System.Web.Mvc.SecurityUtil.ProcessInApplicationTrust(Action action) +22
   System.Web.Mvc.MvcHandler.EndProcessRequest(IAsyncResult asyncResult) +60
   System.Web.Mvc.MvcHandler.System.Web.IHttpAsyncHandler.EndProcessRequest(IAsyncResult result) +9
   System.Web.CallHandlerExecutionStep.System.Web.HttpApplication.IExecutionStep.Execute() +8967885
   System.Web.HttpApplication.ExecuteStep(IExecutionStep step, Boolean& completedSynchronously) +184

Version Information: Microsoft .NET Framework Version:4.0.30319; ASP.NET Version:4.0.30319.272

**Comment by joeddavis** [comment]
I was getting this exception.  I downloaded the source and stepped through the code.  I found that when the "MongoDBMembershipProvider.CreateUser"  method was making several calls to "MongoDBMembershipProvider.GetUser(string, bool)" an exception was being thrown--without stepping into the "GetUser" method.  

Since the two methods in the exception were part of the MongoDB.Drivers.Builders namespace, I figured I'd get known good code in place.  I removed the MongoDB.Bson and MongoDB.Drivers references from the MongoDB.Web project.  Then I re-added the official MongoDB driver from NuGet.... rebuilt all... and it worked like a charm.

Hope that helps someone else.
